### DISCUSSION TITLE
Dynamic "base" property and typescript types

### DISCUSSION THREAD

**Discussion by brpaz** [vuepress/core#965] [Q&A] [answered]
Hello. I have created my `vuepress.config.ts`, since all the request of the project is typescript, but I am having an issue with the type definitions for the `base` property.

I want it to be dynamic, so I am reading from an environment variable like this:

```ts
const baseUrl = process.env.VUEPRESS_BASE_URL || '/';

export default defineUserConfig({
  lang: 'en-US',
  title: 'NestJS starter',
  base: baseUrl,
  description: 'A compreensive starter kit for NestJS applications',
  plugins: [searchPlugin(), backToTopPlugin()],
});
```

It works, but breaks type checking because the `base`  property as a very strict type of ` / ||  /${string}/`

Any way to get around this? Otherwise, I think it would be better to make the type less strict, and be a simple `string`. I think  it´s overkill the way it is defined now.

**Comment by meteorlxy** [comment, marked as answer]
```ts
const baseUrl = (process.env.VUEPRESS_BASE_URL as `/${string}/`) || '/'
```

**Reply by brpaz** [reply]
Ah, nice. Didn´t know I could cast like that. Still pretty new on Typescript.

Thanks.
